src/Operation/DropSearchIndex.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/*
+ * Copyright 2023-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Operation;
+
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\CommandException;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Driver\Server;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnsupportedException;
+
+/**
+ * Operation for the dropSearchIndexes command.
+ *
+ * @see \MongoDB\Collection::dropSearchIndexes()
+ * @see https://mongodb.com/docs/manual/reference/command/dropSearchIndexes/
+ */
+class DropSearchIndex implements Executable
+{
+    private const ERROR_CODE_NAMESPACE_NOT_FOUND = 26;
+
+    private string $databaseName;
+    private string $collectionName;
+    private string $name;
+    private array $options;
+
+    /**
+     * Constructs a dropSearchIndex command.
+     *
+     * @param string                 $databaseName   Database name
+     * @param string                 $collectionName Collection name
+     * @param string                 $name           Index name
+     * @param array{comment?: mixed} $options        Command options
+     * @throws InvalidArgumentException for parameter parsing errors
+     */
+    public function __construct(string $databaseName, string $collectionName, string $name, array $options = [])
+    {
+        if ($name === '') {
+            throw new InvalidArgumentException('Index name cannot be empty');
+        }
+
+        $this->databaseName = $databaseName;
+        $this->collectionName = $collectionName;
+        $this->name = $name;
+        $this->options = $options;
+    }
+
+    /**
+     * Execute the operation.
+     *
+     * @see Executable::execute()
+     * @throws UnsupportedException if write concern is used and unsupported
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function execute(Server $server): void
```

Comment:
One thing I noticed: other drop operations (e.g. DropCollection) return the command result, which can always be ignored by the caller. Did we previously discuss doing that for these operations instead of using a `void` return type?

I know there are some cases where an application might want to access other fields in a command response, such as cluster time, but that might not apply to these Atlas-specific operations.

I'll also note that the other operations support a `typeMap` option specifically to apply to the returned command result; however, I doubt many users (if any at all) make use of that and I wouldn't push for it here.

Reply:
SGTM. If we were implementing the other operations today, I'd be convinced that a `void` return type would be the way to go. This comment was prompted by me thinking of prior art, but that's not an excuse to perpetuate a bad design.